e2e_samples/parking_sensors/scripts/common.sh

```diff
@@ -128,3 +128,45 @@ create_adf_trigger () {
     adfTUrl="${adfFactoryBaseUrl}/triggers/${name}?api-version=${apiVersion}"
     az rest --method put --uri "$adfTUrl" --body @"${ADF_DIR}"/trigger/"${name}".json
 }
+
+# Function to give time for the portal to process the cleanup
+wait_for_process() {
+    local seconds=${1:-15}
+    log "Giving the portal $seconds seconds to process the information..."
+    sleep "$seconds"
+}
+
+cleanup_federated_credentials() {
+    ##Function used in the Clean_up.sh and deploy_azdo_service_connections_azure.sh scripts
+    local sc_id=$1
+    local spnAppObjId=$(az devops service-endpoint show --id "$sc_id" --org "$AZDO_ORGANIZATION_URL" -p "$AZDO_PROJECT" --query "data.appObjectId" -o tsv)
+    local spnCredlist=$(az ad app federated-credential list --id "$spnAppObjId" --query "[].id" -o json)
+    log "Found existing federated credentials. Deleting..."
+
+    # Sometimes the Azure Portal needs a little bit more time to process the information.
+    if [ -z "$spnCredlist" ]; then
+        log "It was not possible to cross Federated credentials and Service Principal.Retrying once more.."
+        wait_for_process
+        local spnCredlist=$(az ad app federated-credential list --id "$spnAppObjId" --query "[].id" -o json)
+        if [ -z "$spnAppObjId" ]; then
```

Comment:
```suggestion
        if [ -z "$spnCredlist" ]; then
```
wasn't this supposed to be the list of credentials?

Reply:
yeahp... msitake of ctrl+c and ctrl+v